client/ipa-getkeytab.c

```diff
@@ -804,6 +890,14 @@ int main(int argc, const char *argv[])
             exit(2);
         }
     }
+    if (server) {
+        ret = ipa_server_to_uri(server, sasl_mech, &ldap_uri);
+        if (ret) {
+            fprintf(stderr, _("Incompatible combination of server name and "
+                              "SASL bind mechanism\n"));
```

Comment:
This error message looks wrong.
ipa_server_to_uri() currently fails only if the mechanism is not recognized (but that one is checked earlier already), or if string allocation fails.